includes/reports/overview/list-table/class-sensei-reports-overview-list-table-courses.php

```diff
@@ -313,4 +319,163 @@ protected function get_additional_filters(): array {
 			'last_activity_date_to'   => $this->get_end_date_and_time(),
 		];
 	}
+
+	/**
+	 * Get total average progress value for courses.
+	 *
+	 * @since  4.5.0
+	 *
+	 * @return float total average progress value for all the courses.
+	 */
+	private function get_total_average_progress(): float {
```

Comment:
This method and other related methods look like a good candidate to put in a separate class.
I think they add too much logic that is irrelevant to the list table, but they are tightly coupled to each other.

Reply:
@merkushin  Hey, so I was thinking about  the best place and naming to move these methods to, so I decided to go with **services** 
Here is the initial commit with the change (commits after this one are just fixing tests and linter errors)
https://github.com/Automattic/sensei/pull/5077/commits/7201c9905e3f7c564bf9b76e3c6e99c6c182a84a